_pytest/main.py

```diff
@@ -499,8 +499,16 @@ def __init__(self, fspath, parent=None, config=None, session=None):
         super(FSCollector, self).__init__(name, parent, config, session)
         self.fspath = fspath
 
+    def _check_initialpaths_for_relpath(self):
```

Comment:
Unfortunately I first merged with `upstream/master` while working here so this PR is bringing #2776 into `features` as well. Because it is a small patch I don't think this will get into the way of reviewing things so if you guys are OK with it I won't bother trying to fix history in this case.